docs/circuit-proof-math.md

```diff
@@ -448,11 +547,227 @@ sends \\({\widetilde{e}}\\) to the verifier, who uses the bottom row
+The prover evaluates polynomials \\(\mathbf{l}(x), \mathbf{r}(x)\\) and performs the [inner product argument](../inner_product_proof/index.html) to prove the relation:
+\\[
+\operatorname{PK}\left\\{
+  ({\mathbf{G}}, {\mathbf{H}}' \in {\mathbb G}^{n}, P', Q \in {\mathbb G}; {\mathbf{l}}, {\mathbf{r}} \in {\mathbb Z\_p}^{n})
+  : P' = {\langle {\mathbf{l}}, {\mathbf{G}} \rangle} + {\langle {\mathbf{r}}, {\mathbf{H}}' \rangle} + {\langle {\mathbf{l}}, {\mathbf{r}} \rangle} Q
+\right\\}
+\\] where \\({\mathbf{H}}' = {\mathbf{y}}^{-n} \circ {\mathbf{H}}\\).
+
+The result of the inner product proof is a list of \\(2k\\) points and \\(2\\) scalars, where \\(k = \log_2(n)\\): \\(\\{L\_k, R\_k, \\dots, L\_1, R\_1, a, b\\}\\).
+
+The complete circuit proof consists of \\(13+2k\\) 32-byte elements:
+\\[
+  \\{A\_I, A\_O, S, T\_1, T\_3, T\_4, T\_5, T\_6, t(x), {\tilde{t}}(x), \tilde{e}, L\_k, R\_k, \\dots, L\_1, R\_1, a, b\\}
+\\]
+
+
+
+Verifier’s algorithm
+--------------------
+
+The input to the verifier is the aggregated proof, which contains the \\(m\\) value commitments \\(V_{(j)}\\),
+and \\(32 \cdot (13 + 2 k)\\) bytes of the proof data where \\(k = \log_2(n)\\) and \\(n\\) is a number of multiplication gates in a circuit:
+
+\\[
+  \\{A\_I, A\_O, S, T\_1, T\_3, T\_4, T\_5, T\_6, t(x), {\tilde{t}}(x), \tilde{e}, L\_k, R\_k, \\dots, L\_1, R\_1, a, b\\}
+\\]
+
+The verifier starts by adding all value commitments \\(V_i\\) to the protocol transcript.
+
+The verifier then [builds constraints](#building-constraints), allocating necessary multiplication gates on the fly and
+generating challenge values bound to the commitments \\(V_i\\).
+
+The verifier uses the Fiat-Shamir transform to obtain challenges by adding the appropriate data sequentially to the protocol transcript:
+
+1. \\(A_I, A_O, S\\) are added to obtain challenge scalars \\(y,z \in {\mathbb Z\_p}\\),
+2. \\(T_1, T_3, T_4, T_5, T_6\\) are added to obtain a challenge \\(x \in {\mathbb Z\_p}\\),
+3. \\(t(x), {\tilde{t}}(x), \tilde{e}\\) are added to obtain a challenge \\(w \in {\mathbb Z\_p}\\).
+
+The verifier flattens constraints:
+
+\\[
+\begin{aligned}
+\mathbf{w}\_L &\gets z \mathbf{z}^q \cdot \mathbf{W}\_L, \\\\
+\mathbf{w}\_R &\gets z \mathbf{z}^q \cdot \mathbf{W}\_R, \\\\
+\mathbf{w}\_O &\gets z \mathbf{z}^q \cdot \mathbf{W}\_O, \\\\
+\mathbf{w}\_V &\gets z \mathbf{z}^q \cdot \mathbf{W}\_V, \\\\
+         w\_c &\gets \langle z \mathbf{z}^q, \mathbf{c} \rangle,
+\end{aligned}
+\\]
+where each of \\(\mathbf{w}\_L, \mathbf{w}\_R, \mathbf{w}\_O\\) has length \\(n\\) and \\(\mathbf{w}\_V\\) has length \\(m\\).
+
+The verifier computes the following scalars for the [inner product argument](../inner_product_proof/index.html):
+
+\\[
+	\\{u\_{1}^{2}, \dots, u\_{k}^{2}, u\_{1}^{-2}, \dots, u\_{k}^{-2}, s_0, \dots, s_{n-1}\\}
+\\]
+
+The goal of the verifier is to check two equations:
+
+1. First, verify the second term of the polynomial \\(t(x)\\) (see [notes](#proving-that-t_2-is-correct)):
```

Comment:
Redundant to have both "1." and "First," (same for "2. Second")

Reply:
Fixed, thanks.